content/influxdb/cloud/reference/json-parsing.md

```diff
@@ -10,45 +10,91 @@ influxdb/v2.0/tags: [mqtt]
 related:
 ---
 
-Use the following examples to help you set up parsing rules for [native subscriptions](/influxdb/cloud/write-data/no-code/native-subscriptions).
+Use the following examples to help you set up JSON parsing rules using [JSON Path](https://jsonpath.com/) 
+for [native subscriptions](/influxdb/cloud/write-data/no-code/native-subscriptions). All JSON paths start with a `$`.
 
-## Example simple MQTT message in JSON format
+## Example MQTT message in "flat" JSON format
 
-```js
+```json
 {
 "device_type":"temperature_sensor",
 "device_id":2036,
 "model_id":"KN24683",
- "temperature":25.0,
- "time":1653998899010000000,
+"temperature":25.0,
+"time":1653998899010000000,
 "error_state":"in_error"
-
 }
 ```
 
-JSON paths start with a “$.” In the above example, all of the values are at the root level of the JSON, so the JSON paths for these elements are very simple:
+With "flat" JSON, all values are at the root level (`$`) and are referenced with dot notation.
+
+| InfluxDB Element | JSON Path       | Data Type | Parsed Result        |
+| :--------------- | :-------------- | :-------- | :------------------- |
+| Measurement      | `$.device_type` | String    | "temperature_sensor" |
+| Timestamp        | `$.time`        | Timestamp | 1653998899010000000  |
+| Tag              | `$.device_id`   | Integer   | 2036                 |
```

Comment:
I don't know if it's worth mentioning, but while this is parsed as an integer, because it's a tag value, it'll be written as a string.

Reply:
Yeah...it's been confusing for users in the UI as well and we want to change. do you have suggestion what to list here in the docs? Maybe "Input Data Type" for clarification?

The UI looks like this:
<img width="878" alt="image" src="https://user-images.githubusercontent.com/32681364/191350384-06601103-8413-4627-963f-3b3e2ea13801.png">

Reply:
With being able to force a data-type in the UI, what's the purpose of the "Data Type" column here. Is it to illustrate what JSON types are parsed as?

Reply:
Yeah just being very hand-holding on what to put in for Data Type. If the data type is incorrect then the field/tag won't be properly parsed in.

Reply:
Ok, it's probably fine as is. The UI does need to be fixed though 😄.